c/interface.c

```diff
@@ -319,6 +327,25 @@ JSBorrowedChar *QTS_GetString(JSContext *ctx, JSValueConst *value) {
   return JS_ToCString(ctx, *value);
 }
 
+JSVoid *QTS_GetArrayBuffer(JSContext *ctx, JSValueConst *data) {
+  size_t length;
+  uint8_t *buffer = JS_GetArrayBuffer(ctx, &length, *data);
+  if (!buffer)
+    return 0;
+  uint8_t *result = malloc(length);
+  if (!result)
+    return result;
+  memcpy(result, buffer, length);
+  return result;
+}
+
+// I don't know how to return two values in C, maybe allocate memory in stack?
+size_t QTS_GetArrayBufferLength(JSContext *ctx, JSValueConst *data) {
```

Comment:
I think we could add `stackAlloc` and use it to pass in a `*size_t` argument like the `&length` arguments in QuickJS's own API, but it's not document anywhere so I think your current approach is fine.

Reply:
Looking online, the best solutions I could find is to use Emscripten's embind API and a separate cpp file to use the API. However, I'm not familiar with Makefile, so I don't really know how to get it to compile the separate cpp file.